### PR TITLE
Fix reset color

### PR DESCRIPTION
**Pull request by lanzhenw**
## What changes are proposed in this pull request?

- Bugfix: `reset` button failed to set `can view` permission user setting back to dataset app_config setting on TEAMS. 
- Previously x-direction scroller shows up unnecessarily on color modal. 

## How is this patch tested? If it is not, please explain why.

(Details)

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: app/packages/core/src/components/ColorModal/ColorFooter.tsx [hunk 1]
```diff
@@ -38,6 +38,11 @@ const ColorFooter: React.FC = () => {
     return true;
   }, [savedSettings]);
 
+  // set to be true only for teams
+  const isTeams = useRecoilValue(fos.compactLayout);
```
Comment (benjaminpkane) [inline app/packages/core/src/components/ColorModal/ColorFooter.tsx:42]: This is logic/naming something we should avoid. `compactLayout` means `isTeams` by coincidence only

File: app/packages/core/src/components/ColorModal/ColorFooter.tsx [hunk 2]
```diff
@@ -47,7 +52,7 @@ const ColorFooter: React.FC = () => {
           text={"Reset"}
           title={`Clear session settings and revert to default settings`}
           onClick={() => {
-            setColorScheme(false, null);
+            setColorScheme(false, isTeams ? datasetDefault : null);
```
Comment (benjaminpkane) [inline app/packages/core/src/components/ColorModal/ColorFooter.tsx:55]: I'm not sure I understand the behavior here. Reset means two different things based on the context? How does a Teams user nullify a dataset default? Ideally we add a `selector` that resolves to the reset value (resolution should be opaque/decoupled)

**Review by benjaminpkane — APPROVED** [review]
LGTM for the release

File: app/packages/core/src/components/ColorModal/ColorFooter.tsx (hunk 2 above)
Comment (lanzhenw) [reply app/packages/core/src/components/ColorModal/ColorFooter.tsx:55]: Reset always means reset to dataset default (if there is one). We have something else to clear dataset default setting. 

By setting colorScheme to `null`, OSS will load dataset default or color pool from `fo.app_config` or `session.config` from the session. But teams can't update through the session. Therefore, I am pointing the setting to datasetDefault here for TEAMS.

Comment (benjaminpkane) [reply app/packages/core/src/components/ColorModal/ColorFooter.tsx:55]: Ok. In my routing work, the color scheme will update through a `session` regardless of what the means to the context (Teams, OSS, or a hypothetical). I'm just advocating that this doesn't leak into OSS and component logic

**Pull request merged**